lib/builders.ncl

```diff
@@ -84,4 +93,75 @@ in
         structured_env.buildInputs = packages,
       }
         | (NickelPkg & { packages | { _ : Derivation } }),
+
+  ShellApplication
+    | doc m%"
+    "%
+    =
+      {
+        name,
+        version | default = "0.0.0",
+        text | NullOr NixString | MutExclusiveWith file "text" "file" | default = null,
+        file | NullOr String | default = null,
+        runtime_inputs | { _ : Derivation } = {},
+        shell_binary | NixString | default = nix-s%"%{lib.import_nix "nixpkgs#bash"}/bin/bash"%,
+
+        build_command = {
+          cmd = nix-s%"%{lib.import_nix "nixpkgs#bash"}/bin/bash"%,
+          args = [
+            "-c",
+            nix-s%"
+              set -euo pipefail
+              source .attrs.sh
+              PATH="%{lib.import_nix "nixpkgs#coreutils"}/bin:$PATH"
```

Comment:
Why not add `coreutils` to `nativeBuildInputs`?

Reply:
`nativeBuildInputs` is handled by `stdenv`, this derivation doesn't use it.